### ISSUE TITLE
Improve UX of adding a third party app as trusted

### ISSUE THREAD

**Issue by michielbdejong** [solid/node-solid-server#1142]
Current UX with NSS 5:
* visit https://pheyvaer.github.io/solid-chess/
* click 'Log In'
* a popup gives you four options, pick 'custom'
* enter your WebId, or at least the protocol+hostname of it
* log in to  your pod  (if  you're not already logged in)
* the popup wil close and the  app will  say you're logged in
* open a separate tab, browse  to your own WebId (make sure to include the  '#me')
* hover over the 'card' title
* select the 'A' icon
* enter the origin of the app, https://pheyvaer.github.io
* tick  the read, write, and append boxes
* click add
* switch back  to  the app pane and start using the app

Proposal: add a checkbox that  makes it easier to add an app as trusted, as part of the login flow. With this, it would become:
* visit https://pheyvaer.github.io/solid-chess/
* click 'Log In'
* a popup gives you four options, pick 'custom'
* enter your WebId, or at least the protocol+hostname of it
* log in to  your pod  (if  you're not already logged in)
* if the app is not in your list of trusted apps yet, you will see three buttons in the popup dialog: 'identity only (don't add as trusted app)', 'identity+read only', and 'identity+read+write'
* click one of those buttons
* the popup wil close and the  app will  say you're logged in
* no need to go into the data browser's trusted  apps pane separately

**Comment by michielbdejong** [comment]
<img width="761" alt="Screenshot 2019-04-17 15 55 19" src="https://user-images.githubusercontent.com/408412/56293390-42931680-6129-11e9-8a7d-88d6b4900e62.png">

**Comment by michielbdejong** [comment]
<img width="900" alt="Screenshot 2019-04-17 15 57 40" src="https://user-images.githubusercontent.com/408412/56293551-8a19a280-6129-11e9-8ec0-f9b18172e515.png">

**Comment by jaxoncreed** [comment]
This makes sense. I'll be sure to try to implement this feature in the new idp.

**Comment by michielbdejong** [comment]
Yeah, we need some cross-dependency there because it means the UI of the IDP needs to trigger a change on the pod.

**Comment by jaxoncreed** [comment]
Ah, that's very true. This would require a whole new dimension where the IDP knows all the pods that a user has, and update each one of them. On top of that, the UI isn't just as simple as "grant access to an app." It would be "Grant access to an app for all of my pods... this one pod... everything excluding this pod?"

**Comment by michielbdejong** [comment]
No, there is only one profile per user, even if there are multiple `pim:storage` pods. You find the user's profile by dereferencing their webid, and that's where their list of trusted apps is. In theory, the user's profile doesn't even have to be on their pod, it could also be generated dynamically from some organization's LDAP server.

**Comment by megoth** [comment]
To inform people following this issue: @jaxoncreed are working on implementing a solution that would allow users to grant access to application as part of the login process _if_ a redirect URL is present _and_ the profile the user tries to authenticate with haven't granted access to application from before _and_ trustedOrigin is set to true for the server.

I'm working on a text that explains what this is about for people uncertain what this part of the login form means.
